Update the Website Setup Instructions in Contributor Docs and README (#49638)
kubernetes/website: README.md

Reviewer: The advice in this file is already pretty much correct; we could improve it if we want, but changes are not strictly needed.

cd website
```

> The Kubernetes website uses the [Docsy Hugo theme](https://github.com/google/docsy#readme) via npm module. Even if you plan to run the website in a container, we strongly recommend pulling in the submodule and other development dependencies by running the following:

Reviewer: ```suggestion
The Kubernetes website uses the [Docsy Hugo theme](https://github.com/google/docsy#readme),
which you can install using NPM. You can also download a ready-made development container image
that includes Hugo and Docsy.

There is also a Git submodule used for tools to generate reference documentation.
```

The Kubernetes website uses the [Docsy Hugo theme](https://github.com/google/docsy#readme) via npm module. Even if you plan to run the website in a container, we strongly recommend pulling in the submodule and other development dependencies by running the following:

### Windows

```powershell

> # fetch submodule dependencies
> git submodule update --init --recursive --depth 1

Reviewer: ```suggestion
# fetch submodule dependencies
git submodule update --init --recursive --depth 1
```

### Linux / other Unix

```bash

> # install dependencies
> npm ci

Reviewer: ```suggestion
# fetch submodule dependencies
make module-init
```

cd website
```

> The Kubernetes website uses the [Docsy Hugo theme](https://github.com/google/docsy#readme),which can be installed via npm. You can also download a pre-configured development container image that includes Hugo and Docsy. Additionally, a Git submodule is used for tools that generate the reference documentation.

Reviewer 2: Wrap lines at 80 characters for ease of reviews :)

> The Kubernetes website uses the [Docsy Hugo theme](https://github.com/google/docsy#readme),which can be installed via npm. You can also download a pre-configured development container image that includes Hugo and Docsy. Additionally, a Git submodule is used for tools that generate the reference documentation.

Reviewer 2: @sftim wasn't there a recent change to remove the Hugo dependency from the container image?
